pkg/kubelet/kubelet_server_journal.go

```diff
@@ -0,0 +1,390 @@
+		boot, err = strconv.Atoi(bootValue)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("boot"), bootValue, err.Error()))
+		} else {
+			nlq.Boot = &boot
+		}
+	}
+
+	var tailLines int
+	tailLinesValue := query.Get("tailLines")
+	if len(tailLinesValue) > 0 {
+		tailLines, err = strconv.Atoi(tailLinesValue)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("tailLines"), tailLinesValue, err.Error()))
+		} else {
+			nlq.TailLines = &tailLines
+		}
+	}
+
+	if len(allErrs) > 0 {
+		return nil, allErrs
+	}
+
+	return nlq, allErrs
+}
+
+func validateServices(services []string) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	var totalLength int
+	for _, s := range services {
+		totalLength += len(s)
+		if err := safeString(s); err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("query"), s, err.Error()))
+		}
+	}
+
+	maxTotalServiceNamesLength := 1000
+	if totalLength > maxTotalServiceNamesLength {
+		allErrs = append(allErrs, field.TooLongMaxLength(field.NewPath("query"), services, maxTotalServiceNamesLength))
+	}
+	return allErrs
+}
+
+func (n *nodeLogQuery) validate() field.ErrorList {
+	// This is to account for the case that query is empty and the caller is directly trying to query a file
+	// Example: kubectl get --raw /api/v1/nodes/$node/proxy/logs/foo.log
+	if len(n.Services) == 0 && len(n.Files) == 0 && n.options == (options{}) {
+		return field.ErrorList{}
+	}
+
+	allErrs := validateServices(n.Services)
+	switch {
+	case len(n.Files) == 0 && len(n.Services) == 0 && n.options != (options{}):
+		allErrs = append(allErrs, field.Required(field.NewPath("query"), "cannot be empty with options"))
+	case len(n.Files) > 0 && len(n.Services) > 0:
+		allErrs = append(allErrs, field.Invalid(field.NewPath("query"), fmt.Sprintf("%v, %v", n.Files, n.Services),
+			"cannot specify a file and service"))
+	case len(n.Files) > 1:
+		allErrs = append(allErrs, field.Invalid(field.NewPath("query"), n.Files, "cannot specify more than one file"))
+	case len(n.Files) == 1 && n.options != (options{}):
+		allErrs = append(allErrs, field.Invalid(field.NewPath("query"), n.Files, "cannot specify file with options"))
+	}
+
+	if n.SinceTime != nil && n.UntilTime != nil && (n.SinceTime.After(*n.UntilTime)) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("untilTime"), n.UntilTime, "must be after `sinceTime`"))
+	}
+
+	if n.Boot != nil && runtime.GOOS == "windows" {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("boot"), *n.Boot, "boot is not supported on Windows"))
+	}
+
+	if n.Boot != nil && *n.Boot > 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("boot"), *n.Boot, "must be less than 1"))
+	}
+
+	if n.TailLines != nil {
+		if err := utilvalidation.IsInRange((int)(*n.TailLines), 0, 100000); err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("tailLines"), *n.TailLines, err[0]))
+		}
+	}
+
+	if err := safeString(n.Pattern); err != nil {
```

Comment:
I don't see us validating Files or passing them through safeString

Reply:
To validate files, I am now calling [os.Stat](https://github.com/kubernetes/kubernetes/pull/96120/commits/72a60d106a1ffcefa093f93931a8454ee6a56656#diff-17cea30c20aab85b597ee2d641ea841db6b1812b3ac22316e312566c3f6edc3bR225).